### DISCUSSION TITLE
Evolving Dirichlet and Neumann boundary with respect to time

### DISCUSSION THREAD

**Discussion by sahu15** [idaholab/moose#15783] [Q&A General] [answered]
Dear MOOSE team,

I am using KKS example in phase field module to simulate microstructure evolution. I have a generated 3D mesh in MOOSE with a user-defined sideset ( the red patch). I have applied dirichlet boundary on the red patch and Neumann boundary condition ( Flux=0 ) everywhere else in the boundary (top, bottom, left, right , front, back).
Now during my simulation, I want to change the radius of the red patch as a function of time depending upon the evolution of field variables within my system. Could you provide any suggestions to accomplish this?
![image](https://user-images.githubusercontent.com/68827145/91067359-ceac2c80-e600-11ea-90d0-db05818c06ba.png)

Regards,
Sarita

**Comment by YaqiWang** [comment, marked as answer]
You can create a new integrated BC that on the entire boundary. For the points within the radius, you can apply the Dirichlet condition weakly (check out PenaltyDirichletBC). And you apply Neumann BC elsewhere. You can make radius as a function of time.

**Reply by sahu15** [reply]
Hi Yaqi,

Thank you for the suggestion. I will try it out. 

Regards,
Sarita

**Reply by sahu15** [reply]
Hi Yaqi,
I am creating "ADMixedDirichletBC" class to enforce the boundary condition I described earlier. In the computeQpResidual() function, I  need the x and y coordinates of the current quadrature point(_qp). I know my x<sub>o</sub>, y<sub>o</sub> and r<sub>o</sub>.   **How to extract x and y co-ordinates from _qp object ?** I tried to refer to the workshop video, but I could not find it.
Regards,
Sarita
![image](https://user-images.githubusercontent.com/68827145/91195872-6b390200-e6c7-11ea-91d1-0cd87701af1f.png)

**Reply by sahu15** [reply]
Also, I am using "ADPenaltyDirichlet.C" as my template. So _p and _v are my penalty factor and boundary value, repectively.

**Reply by YaqiWang** [reply]
`_q_point[_qp]` holds the coordinates of the quadrature point. `_normals[_qp]` is for outward unit norm, which could be useful for you. `_current_boundary_id` is the boundary ID of the boundary.

**Reply by sahu15** [reply]
Thank you, Yaqi!
